### PR TITLE
Fix conv weight scale mask issue with onednn 3.0

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
**Summary**
Fix weight scale mask issue of convolution with groups > 1. Align with latest onednn 3.0. The issue was found by PyTorch UT with stock PyTorch + latest onednn 3.0. The issue does not exist with some older version of onednn 3.0.

For convolution with groups > 1, we need to check if the length of scale vector is equal to (1) number of OC or (2) OC * groups. In the first case, the mask should be 2, corresponding to the second dimension of format `goi...`. In the second case, the mask should be 3, corresponding to both the first and second dimensions of format `goi...`. Things are similar for deconv, with one difference that the format is `gio...`

This change won't affect IPEX since IPEX does not use ideep for quantization.

The fix is for PyTorch but we plan to merge it to the ideep branch that IPEX uses since we don't have a plan to update onednn 3.0 in stock PyTorch right now.

**Test plan**
PyTorch UT for quantization with stock PyTorch + latest onednn 3.0

@jgong5 @XiaobingSuper @yanbing-j @Guobing-Chen @leslie-fang-intel Please review. Thanks!

### CODE REVIEW

File: include/ideep/utils.hpp [hunk 1]
```diff
@@ -280,12 +280,47 @@ inline int tensor_scale_mask(dim scale_size, bool grouped) {
...
+        // scales apply on o (3rd dimension)
+        return 4;
+      } else {
+        // scale_size = groups * channels_per_group
+        // scales apply on g + o (1st and 3rd dimensions)
+        return 5;
+      }
+    }
+    else {
+      // format = goiw/giohw/giodhw for 1/2/3d
```
Comment (leslie-fang-intel) [inline include/ideep/utils.hpp]: nit. format should be `goiw/goihw/goidhw` for conv?

Comment (Xia-Weiwen) [reply include/ideep/utils.hpp]: Thanks. The typo is fixed

File: include/ideep/operators/conv.hpp [hunk 2]
```diff
@@ -129,7 +129,11 @@ struct conv_deconv_utils {
       // Set scales to op_attr
       int src_scale_mask = 0;
       op_attr.set_scales(DNNL_ARG_SRC, src_scale_mask, {1.0f / src_scales_in[0]});
-      auto wei_scale_mask = utils::conv_weight_scale_mask(weights_scales_in.size(), groups > 1, is_deconv);
+      // tensor::get_dim(0) returns internal_dims[0] * internal_dims[1] for grouped weight
+      // i.e., g * oc_per_group for conv and g * ic_per_group for deconv
+      int oc_per_group = is_deconv ? weight_grouped.get_dim(1) : weight_grouped.get_dim(0) / groups;
```
Comment (jgong5) [inline include/ideep/operators/conv.hpp:139]: should you divide `groups` on `weight_grouped.get_dim(1)` too?

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: include/ideep/operators/conv.hpp (hunk 2 above)
Comment (Xia-Weiwen) [reply include/ideep/operators/conv.hpp:139]: No. For grouped deconv, weight format is `gio...`. `get_dim(0)` returns `g * i` and `get_dim(1)` returns `o`, which is desired.

Comment (jgong5) [reply include/ideep/operators/conv.hpp:139]: Can you clarify this a bit in the comment? The current comment doesn't read straightforward to me.

**Review by jgong5 — APPROVED** [review]

File: include/ideep/operators/conv.hpp (hunk 2 above)
Comment (Xia-Weiwen) [reply include/ideep/operators/conv.hpp:139]: OK. I have added more details in the comment.

**Comment by Xia-Weiwen** [comment]
Hi @yanbing-j @XiaobingSuper Could you help merge this? Thanks!

**Pull request merged**